docs/arcaflow/workflows/flow-control.md

```diff
@@ -107,3 +107,179 @@ outputs:
+    input:
+      name: !expr $.input.name
+
+outputs:
+  success:
+    name: !expr $.steps.example.outputs.success
+```
+
+The input variable `name`'s value is repeated across for each iteration in this input.
+
+```yaml title="input.yaml"
+repeated_inputs: 
+  name: Matt
+iterations:
+  - loop_id: 1
+  - loop_id: 2
+  - loop_id: 3
+  - loop_id: 4
+```
+Using `bindConstants()` we can factor out `name` into its own subsection, here we have named our new subsection `repeated_inputs`.
+
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        repeated_inputs:
+          type:
+            type_id: ref
+            id: RepeatedValues
+        iterations:
+          type:
+            type_id: list
+            items:
+              id: SubRootObject
+              type_id: ref
+    RepeatedValues:
+      id: RepeatedValues
+      properties:
+        name:
+          type:
+            type_id: string
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer            
+            
+steps:
+  foreach_loop:
+    kind: foreach
+    items: !expr 'bindConstants($.input.iterations, $.input.repeated_inputs)'
+    workflow: subworkflow.yaml
+    parallelism: 1
+
+outputs:
+  success:
+    fab_four: !expr $.steps.foreach_loop.outputs.success.data
+```
+
+```yaml title="subworkflow.yaml"
+version: v0.2.0
+input:
+  root: SubRootObject__RepeatedValues
+  objects:
+    SubRootObject__RepeatedValues:
+      id: SubRootObject__RepeatedValues
+      properties:
+        constant:
+          type:
+            type_id: ref
+            id: RepeatedValues
+        item:
+          type:
+            type_id: ref
+            id: SubRootObject
+    RepeatedValues:
+      id: RepeatedValues
+      properties:
+        name:
+          type:
+            type_id: string
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer
+
+steps:
+  example:
+    plugin:
+      deployment_type: image
+      src: quay.io/arcalot/arcaflow-plugin-template-python::all_d5dd9d6
+    input:
+      name: !expr $.input.constant.name
+
+outputs:
+  success:
+    loop_id: !expr $.input.item.loop_id
+    beatle: !expr $.steps.example.outputs.success
+```
+
+To use the generated values from `bindConstants()`, a new schema representing these bound values must be added to the input schema section of our `subworkflow.yaml`, `input`. This new schema's ID will be the ID of the schema that defines the items in your list, in this case `SubRootObject` and the schema name that defines your repeated inputs, in this case `RepeatedValues`, concatenated with a double underscore, `__`. This creates our new schema ID, `SubRootObject__RepeatedValues`. You are required to use this schema ID because it is generated from the names of your other schemas.
```

Comment:
This might be better pushed up as well, so that we read the description of this weird jumbled new schema ID *before* we see the actual schema in the workflow.

Reply:
did it